src/osgEarth/MPScreenSpaceLayoutSG.cpp

```diff
@@ -482,14 +494,80 @@ struct /*internal*/ MPDeclutterSortSG : public osgUtil::RenderBin::SortCallback
                 updateOffsetForAutoLabelOnLine(box, vp, annoDrawable->_cull_anchorOnScreen, annoDrawable, camVPW, slidingOffset, to);
                 annoDrawable->_cull_anchorOnScreen += slidingOffset;
             }
-            
+
+            // Computes label location for the grid mora (visible part of the polygon)
+            if (annoDrawable->polygonVisible())
+            {
+                double maxAlt = annoDrawable->getTextPolygonAltitude();
+                double currentAltitude;
+                cam->getUserValue("altitude", currentAltitude);
+
+                //                std::cout <<"alt: ,"<<alt << " current alt: "<< currentAltitude<<"\n";
+                if (currentAltitude > maxAlt)
+                {
+                    visible = true;
+                }
+                else
+                {
+                    visible = false;
+                    // use the centroid to display the label if visible
+                    osg::Vec3d centroid = annoDrawable->getAnchorPoint() * MVP;
+                    double lim = 1.0;
+                    if (centroid.x() > -lim && centroid.x() < lim &&
+                            centroid.y() > -lim && centroid.y() < lim )
+                    {
+                        visible = true;
+                    }
+                    else
+                    {
+                        osg::Vec3d pc[5];
+                        pc[0] = annoDrawable->getLineStartPoint()* MVP;
+                        pc[2] = annoDrawable->getLineEndPoint()* MVP;
+                        pc[1] = annoDrawable->getPolygonPoint1()* MVP;
+                        pc[3] = annoDrawable->getPolygonPoint2()* MVP;
+                        pc[4] = pc[0];      // to close the polygon (repeat 1st point)
+
+                        boost_polygon geomMora;
```

Comment:
mora is a specific case (you can say it if you want in the comments) but don't name a variable *mora*